lightning/src/util/sweep.rs

```diff
@@ -356,6 +357,60 @@ where
 	logger: L,
 }
 
+pub struct OutputSweeperSync<B: Deref, D: Deref, E: Deref, F: Deref, K: Deref, L: Deref, O: Deref>
+where
+	B::Target: BroadcasterInterface,
+	D::Target: ChangeDestinationSource,
+	E::Target: FeeEstimator,
+	F::Target: Filter + Sync + Send,
+	K::Target: KVStore,
+	L::Target: Logger,
+	O::Target: OutputSpender,
+{
+	sweeper: OutputSweeper<B, D, E, F, K, L, O>,
+}
+
+impl<B: Deref, D: Deref, E: Deref, F: Deref, K: Deref, L: Deref, O: Deref>
```

Comment:
What might be good about this wrapper is that it eliminates the possibility of someone implementing async logic in combination with future poll/ready checking.